src/pages/address/components/address_field/AddressField.jsx

```diff
@@ -0,0 +1,254 @@
+  const { selectedAddress } = useAddress();
+
+  const [formData, setFormData] = useState(
+    selectedAddress || {
+      name: "",
+      mobile: "",
+      address: "",
+      pin: "",
+      city: "",
+      state: "",
+      landmark: "",
+    }
+  );
+
+  const [formValidation, setFormValidation] = useState({
+    name: true,
+    mobile: true,
+    address: true,
+    pin: true,
+    city: true,
+    state: true,
+    landmark: true,
+  });
+
+  const [validationMsg, setValidationMsg] = useState({
+    name: "",
+    mobile: "Please fill your mobile",
+    address: "",
+    pin: "",
+    city: "",
+    state: "",
+    landmark: "",
+  });
+
+  const { addToAddressList, removeFromAddressList } = useAddress();
+
+  const { name, mobile, address, pin, city, state, landmark } = formData;
+
+  const updateFormData = (target, key) => {
+    setFormData((formData) => ({ ...formData, [key]: target.value }));
+  };
+
+  const submitForm = async () => {
+    try {
+      // To check if each field in address has valid entry
+      ["name", "mobile", "address", "pin", "city", "state", "landmark"].map(
+        (field) =>
+          formData[field].length > 0
+            ? setFormValidation((validation) => ({
+                ...validation,
+                [field]: true,
+              }))
+            : setFormValidation((validation) => ({
+                ...validation,
+                [field]: false,
+              }))
+      );
+
+      // To check if all the fields are not empty
+      if (
+        name.length > 1 &&
+        mobile.length === 10 &&
+        address.length > 1 &&
+        pin.length > 1 &&
+        city.length > 1 &&
+        state.length > 1 &&
+        landmark.length > 1
+      ) {
+        const addressData = {
+          _id: uuid(),
+          name: name,
+          mobile: mobile,
+          address: address,
+          pin: pin,
+          city: city,
+          state: state,
+          landmark: landmark,
+        };
+
+        // To check if its a new entry or and update
+
+        if (selectedAddress.name === "" && selectedAddress.address === "") {
+          // Add address
+          const res = await axios.post(
+            "/api/user/addresses",
+            {
+              address: addressData,
+            },
+            {
+              headers: { authorization: getUserToken() },
+            }
+          );
+
+          if (res?.status === 201) {
+            addToAddressList(addressData);
+          }
+        } else {
+          // Update address
+          const res = await axios.post(
+            `/api/user/addresses/${selectedAddress._id}`,
+            {
+              address: { ...addressData, _id: selectedAddress._id }, // To make _id consistent through out updates
+            },
+            {
+              headers: { authorization: getUserToken() },
+            }
+          );
+          console.log(res);
+          if (res?.status === 200) {
```

Comment:
I don't think we need to use optional chaining here. We send data successfully we always get a response.

Reply:
Ok, got it